### PR TITLE
Feat/implement back to top button

### LINKED ISSUE #2662: Implement "Back to top" feature. Improve UX when scrolling longer content.
**Story:** Scrolling longer content UX needs improvement. 2 issues identified:
1. It's difficult for the user to smoothly go back to the top of the scrollable item lists and longer content pages (transactions, tokens, accounts, NFTs etc. but also Terms and Privacy pages). 
**UX/UI solution:** To fix this we need to introduce back to top feature.

2. It's not clearly visible to the users when they reach the end of longer item lists (for example transactions history list) that there is no more scrollable content. 
**UX/UI solution:** 40px padding to be added to the bottom of all scrollable items lists.

**Acceptance criteria:** 
1. Add a "back to top" button feature after design provided below. The button should appear for scrollable content longer/higher than 800px after this point is reached (visible in the viewport) while user is scrolling. Apply smooth animations and transitions to the elements when auto-scrolling back to top after the button is clicked. 

2. Apply 40px bottom padding to all scrollable content where the Back to top button is displayed.
- Transactions;
- Assets, 
- Account selector (when more accounts are available in the list);
- NFTs (on the Roadmap);
- Terms;
- Privacy;
- any longer content that needs scrolling.

**Design:** 

Back to top button:
Image
Image

Bottom padding:
Image

**Figma reference:** https://www.figma.com/file/3oGLWzSH0oJljo4RETZtur/Superhero-Wallet-UI-(%E2%9C%94%EF%B8%8FUpdated)?type=design&node-id=29676-167098&mode=design&t=5TWL8qAfX9jKDV39-0

**Prototype recording:**

https://github.com/superhero-com/superhero-wallet/assets/12225410/df34da84-1b63-458b-aa56-ac6797a30462

### PR DESCRIPTION
**Pull request by martinkaintas**
closes #2662 

Based on `feat/eth-support` in order to apply changes to the ETH asset list component.
Turns out we do not need the `is end reached` variable since we want the `back to top` button to appear even if the user has not reached the end yet.
One small improvement we can do once we have the `is end reached` variable is that we can hide the bottom padding that is added by the `back to top` button and only show it when `is end reached`

### CODE REVIEW

File: src/popup/components/BackToTop.vue [hunk 1]
```diff
@@ -1,78 +1,143 @@
...
   components: {
     BtnPlain,
     Chevron,
   },
-  data() {
-    return {
-      isVisible: false,
-    };
+  props: {
+    sticky: { type: Boolean, default: false },
```
Comment (peronczyk) [inline src/popup/components/BackToTop.vue:40]:
```suggestion
    sticky: Boolean,
```

File: src/popup/components/BackToTop.vue [hunk 2]
```diff
@@ -1,78 +1,143 @@
...
+});
 </script>
 
 <style lang="scss" scoped>
 @use '../../styles/variables';
 @use '../../styles/mixins';
 
 .back-to-top {
-  @include mixins.flex(center, center);
+  .back-to-top-spacer {
```
Comment (peronczyk) [inline src/popup/components/BackToTop.vue:84]:
```suggestion
  &-spacer {
```

File: src/popup/components/BackToTop.vue [hunk 3]
```diff
@@ -1,78 +1,143 @@
...
+    width: 100%;
+    height: 40px;
+  }
+
+  .back-to-top-btn-container {
```
Comment (peronczyk) [inline src/popup/components/BackToTop.vue:89]:
```suggestion
  &-btn-container {
```

File: src/popup/components/BackToTop.vue [hunk 4]
```diff
@@ -1,78 +1,143 @@
...
+      linear-gradient(
+        180deg,
+        rgba(variables.$color-bg-4, 0) 5%,
+        rgba(variables.$color-bg-4, 0.8) 60%,
+        rgba(variables.$color-bg-4, 0.9) 100%
+      );
 
-  &:hover {
-    background: rgba(variables.$color-white, 0.08);
+    .back-to-top-btn {
```
Comment (peronczyk) [inline src/popup/components/BackToTop.vue:108]:
```suggestion
    &-btn {
```

File: src/popup/components/BackToTop.vue [hunk 5]
```diff
@@ -1,78 +1,143 @@
...
         top: 0,
         behavior: 'smooth',
       });
-    },
+    }
+
+    onMounted(() => {
+      if (el.value?.parentNode) {
+        el.value?.parentNode.addEventListener('scroll', handleVisibility);
+      }
```
Comment (peronczyk) [inline src/popup/components/BackToTop.vue:61]:
```suggestion
      el.value?.parentNode?.addEventListener('scroll', handleVisibility);
```

File: src/popup/components/BackToTop.vue [hunk 6]
```diff
@@ -1,78 +1,143 @@
...
+    });
+
+    onBeforeUnmount(() => {
+      if (el.value?.parentNode) {
+        el.value?.parentNode.removeEventListener('scroll', handleVisibility);
+      }
```
Comment (peronczyk) [inline src/popup/components/BackToTop.vue:67]:
```suggestion
      el.value?.parentNode?.removeEventListener('scroll', handleVisibility);
```

File: src/popup/components/BackToTop.vue (hunk 4 above)
Comment (martinkaintas) [reply src/popup/components/BackToTop.vue:108]: That would produce the following class name: `back-to-top-btn-container-btn`

**Comment by Liubov-crypto** [comment]
LGTM

**Pull request merged**